packages/snaps-sdk/src/types/caip.ts

```diff
@@ -22,3 +33,85 @@ export type ChainId = CaipChainId;
  * @deprecated Use {@link CaipAccountId} instead.
  */
 export type AccountId = CaipAccountId;
+
+/**
+ * A struct representing a list of CAIP-10 account IDs where the account addresses and namespaces are the same.
+ */
+export const CaipAccountIdsMatchedByAddressAndNamespaceStruct = refine(
+  array(CaipAccountIdStruct),
+  'Matching Addresses Account ID List',
+  (value) => {
+    const parsedAccountIds = value.map((accountId) =>
+      parseCaipAccountId(accountId),
+    );
+
+    if (
+      !parsedAccountIds.every(
+        ({ address, chain: { namespace } }) =>
+          address === parsedAccountIds[0].address &&
+          namespace === parsedAccountIds[0].chain.namespace,
+      )
+    ) {
+      return 'All account IDs must have the same address and namespace.';
+    }
+
+    return true;
+  },
+);
+
+/**
+ * A struct representing a list of non-EIP-155 CAIP-10 account IDs where the account addresses are the same.
+ */
+export const NonEip155CaipAccountIdsMatchedByAddressAndNamespaceStruct = refine(
+  CaipAccountIdsMatchedByAddressAndNamespaceStruct,
```

Comment:
Maybe simpler and more performant to combine the two structs in use here? We don't seem to be using `CaipAccountIdsMatchedByAddressAndNamespaceStruct` on its own